app/jobs/hyrax/revoke_edit_from_members_job.rb

```diff
@@ -0,0 +1,15 @@
+module Hyrax
+  # Revokes edit access for the supplied user for the members attached to a work
+  class RevokeEditFromMembersJob < ApplicationJob
+    queue_as Hyrax.config.ingest_queue_name
+
+    # @param [ActiveFedora::Base] work - the work object
+    # @param [String] user_key - the user to remove
+    def perform(work, user_key)
+      # Iterate over ids because reifying objects is slow.
+      work.member_ids.each do |file_set_id|
+        RevokeEditJob.perform_now(file_set_id, user_key)
```

Comment:
@darrenleeweber There is no way in ActiveFedora to use a transaction (mostly because there is no support for transactions in Solr).